Specs/Scene/ImplicitSubtreeSpec.js

```diff
@@ -0,0 +1,378 @@
+    },
+  });
+
+  it("gets availability from internal buffer", function () {
+    var subtreeDescription = {
+      tileAvailability: {
+        descriptor: "11010",
+        lengthBits: 5,
+        isInternal: true,
+      },
+      contentAvailability: {
+        descriptor: "11000",
+        lengthBits: 5,
+        isInternal: true,
+      },
+      childSubtreeAvailability: {
+        descriptor: "1111000010100000",
+        lengthBits: 16,
+        isInternal: true,
+      },
+    };
+
+    var results = ImplicitTilingTester.generateSubtreeBuffers(
+      subtreeDescription
+    );
+    var subtree = new ImplicitSubtree(
+      subtreeResource,
+      results.subtreeBuffer,
+      implicitQuadtree
+    );
+    return subtree.readyPromise.then(function () {
+      expectTileAvailability(subtree, subtreeDescription.tileAvailability);
+      expectContentAvailability(
+        subtree,
+        subtreeDescription.contentAvailability
+      );
+      expectChildSubtreeAvailability(
+        subtree,
+        subtreeDescription.childSubtreeAvailability
+      );
+    });
+  });
+
+  it("gets availability from external buffer", function () {
+    var subtreeDescription = {
+      tileAvailability: {
+        descriptor: "11010",
+        lengthBits: 5,
+        isInternal: false,
+      },
+      contentAvailability: {
+        descriptor: "11000",
+        lengthBits: 5,
+        isInternal: false,
+      },
+      childSubtreeAvailability: {
+        descriptor: "1111000010100000",
+        lengthBits: 16,
+        isInternal: false,
+      },
+    };
+    var results = ImplicitTilingTester.generateSubtreeBuffers(
+      subtreeDescription
+    );
+
+    spyOn(Resource.prototype, "fetchArrayBuffer").and.returnValue(
+      when.resolve(results.externalBuffer)
+    );
+    var subtree = new ImplicitSubtree(
+      subtreeResource,
+      results.subtreeBuffer,
+      implicitQuadtree
+    );
+    return subtree.readyPromise.then(function () {
+      expectTileAvailability(subtree, subtreeDescription.tileAvailability);
+      expectContentAvailability(
+        subtree,
+        subtreeDescription.contentAvailability
+      );
+      expectChildSubtreeAvailability(
+        subtree,
+        subtreeDescription.childSubtreeAvailability
+      );
+    });
+  });
+
+  it("tile and content availability can share the same buffer", function () {
+    var subtreeDescription = {
+      tileAvailability: {
+        descriptor: "11010",
+        lengthBits: 5,
+        isInternal: false,
+      },
+      contentAvailability: {
+        shareBuffer: true,
+        descriptor: "11010",
+        lengthBits: 5,
+        isInternal: false,
+      },
+      childSubtreeAvailability: {
+        descriptor: "1111000010100000",
+        lengthBits: 16,
+        isInternal: false,
+      },
+    };
+    var results = ImplicitTilingTester.generateSubtreeBuffers(
+      subtreeDescription
+    );
+
+    spyOn(Resource.prototype, "fetchArrayBuffer").and.returnValue(
```

Comment:
Since this test is checking that the buffer is only loaded once it would be good to check how many times this spy is called.